### PR TITLE
Admin API end point added for user reset role and password

### PR DESCRIPTION
**Pull request by ashish-jabble**
- [x] reset => foglamp/admin/{user_id}/reset ( Admin  route to reset Password and role for a user)

- [x] delete_user => /foglamp/admin/{user_id}/delete (Only Admin can delete the user) 

- [x] create_user => /foglamp/admin/user (Only Admin can create the user) 

- [x] update_password => foglamp/user/{username}/password (Normal user can update the password)

- [ ] update_user =>  /foglamp/user/{id} (For normal user can update the profile) - Right now not implemeted as FOGL-1226 pending

**NOTE** -- will update tests once approved :)

### CODE REVIEW

File: python/foglamp/services/core/api/auth.py [hunk 1]
```diff
@@ -371,6 +326,64 @@
...
+    if password and not isinstance(password, str):
+        _logger.warning(PASSWORD_ERROR_MSG)
+        raise web.HTTPBadRequest(reason=PASSWORD_ERROR_MSG)
+    if password and not re.match(PASSWORD_REGEX_PATTERN, password):
+        _logger.warning(PASSWORD_ERROR_MSG)
+        raise web.HTTPBadRequest(reason=PASSWORD_ERROR_MSG)
+
+    user_data = {}
+    if 'role_id' in data:
+        user_data.update({'role_id': data['role_id']})
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/auth.py]: what if a bad role id (non existent) is passed

File: python/foglamp/services/core/api/auth.py [hunk 2]
```diff
@@ -32,6 +32,8 @@
     
     | POST                       | /foglamp/login                                      |
     | PUT                        | /foglamp/{user_id}/logout                           |
+    
+    | PUT                        | /foglamp/admin/{user_id}/reset                      |
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/auth.py]: please add /admin for create and delete user too

File: python/foglamp/services/core/api/auth.py [hunk 3]
```diff
@@ -371,6 +327,68 @@
...
+    password = data.get('password')
+    role_id = data.get('role_id')
+
+    if not role_id and not password:
+        msg = "Nothing to update the user"
+        _logger.warning(msg)
+        raise web.HTTPBadRequest(reason=msg)
+
+    if not is_valid_role(role_id):
+        _logger.warning("Create user requested with bad role id")
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/auth.py]: create user? its for reset password or update role here

Comment (praveen-garg) [reply python/foglamp/services/core/api/auth.py]: what if no role_id passed; the check above only handles AND case

**Comment by ashish-jabble** [comment]
@praveen-garg made changes as per your suggestions.

**Pull request merged**